src/kernel/src/process/session.rs

```diff
@@ -1,18 +1,41 @@
-use std::num::NonZeroI32;
+use crate::{fs::Vnode, tty::Tty};
+use gmtx::*;
+use std::{num::NonZeroI32, sync::Arc};
 
 /// An implementation of `session` structure.
 #[derive(Debug)]
 pub struct VSession {
-    id: NonZeroI32, // s_sid
-    login: String,  // s_login
+    id: NonZeroI32,               // s_sid
+    login: Gutex<String>,         // s_login
+    vnode: Option<Vnode>,         // s_ttyvp
```

Comment:
Do you know why FreeBSD keep this?

Reply:
I don't

Reply:
How this field is set?

Reply:
"Vnode of controlling tty".

Reply:
http://fxr.watson.org/fxr/source/fs/devfs/devfs_vnops.c?v=FREEBSD-9-STABLE#L776

Reply:
In that case better to merge those fields together so it will be behind a single `Option`.

Reply:
You mean s_ttydp and s_ttyvp? A s_ttydp equivalent isn't in our code currently.

Reply:
I mean `vnode` and `tty` because it seems like both of them must be `Some` or `None` at a given time.

Reply:
<img width="554" alt="image" src="https://github.com/obhq/obliteration/assets/48620541/c426c0d0-a442-4354-93b0-024b99df8025">
Is this ok? I don't know how else to do this. Having it like this makes it map very weirdly to the FreeBSD code.

Reply:
Sometime it will look weird to FreeBSD code but it is much more fitted with Rust so don't worry about that. In the end we will not care if our code similar to FreeBSD or not as long as it archive the same behavior.

Reply:
The current code now panics though :D

Reply:
Actually. I don' really think the fields should be together. in one option. 
<img width="578" alt="image" src="https://github.com/obhq/obliteration/assets/48620541/f8938bf7-a80e-4c3a-af6c-28ea81e4009a">
The first line triggers the tty ioctl and only after that, the vnode is set on the second-to-last line. But the tty ioctl could return an error and the vnode won't be set in that case.

Reply:
You mean the TTY will be set alone in the d_ioctl?

Reply:
Actually, no. I'm just dumb. I think I understand what needs to be done. But even with that, I'll have to wait for the fs refactor to finalize, because I don't have access to the vnode right now because we're not there yet with the fs rewrite

Reply:
Probably use `Fs::root` for a vnode for now then replace it later.

Reply:
So I have to pass Arc<Fs> all the way down to tty? :(

Reply:
Okay I see what you mean. In this case just remove vnode from here should be okay.